### ISSUE TITLE
Merging variables with overlapping not but not conflicting values

### ISSUE THREAD

**Issue by shoyer** [pydata/xarray#835]
It should be possible to safely merge together variables with values `[NaN, 1, 2]` and `[0, 1, NaN]` by using methods such as `combine_first` (which should also be OK with conflicting values, like the pandas method) and `merge` (which should raise if values conflict).

See this stackoverflow post for a merge example: http://stackoverflow.com/questions/36731870/how-to-merge-xarray-datasets-with-conflicting-coordinates

**Comment by shoyer** [comment]
I'm currently in the process of refactoring the merge code so this should be pretty easy to add in.

**Comment by shoyer** [comment]
Actually, we can just write:
```
def combine_first(left, right):
    return ops.where(ops.notnull(left), left, right)
```

**Comment by shoyer** [comment]
Fixed by #1204
